lang/javascript/javascript.py

```diff
@@ -6,6 +6,38 @@
 tag: user.javascript
 """
 
+ctx.lists["user.code_common_function"] = {
+    "entries": "Object.entries",
+    "flatten": "flatten",
+    "from entries": "Object.fromEntries",
+    "keys": "Object.keys",
+    "max": "Math.max",
+    "min": "Math.min",
+    "abs": "Math.abs",
+    "round": "Math.round",
+    "floor": "Math.floor",
+    "print": "console.log",
+    "values": "Object.values",
+}
+
+mod.list("code_chain_function", "Function to use in a dotted chain, eg .foo()")
```

Comment:
I noticed C# puts `.ToString` in the `code_common_function` list. Maybe doing the same here would be fine?

Reply:
Yeah I guess we could do it that way?  These chained functions are currently accessed using "state", so I was thinking to stick with that approach

Reply:
Fwiw I personally use "chain" for these

Reply:
I think of them a bit differently from "funk".  For example, I'm not sure it makes sense to wrap with them, etc

Reply:
@nriley and I decided to use "chain" for this one, and then add deprecation notifications on the old "state" ones

Reply:
I wonder if chain is the right word. I'd be inclined to call it meth, short for method.

Reply:
"meth" is...interesting 😅. Not sure that's gonna be the best word to be spamming in a workplace setting, but that aside, it's not a bad term 🤔

I'd also be inclined in the future to also use the same word for things like `.length`, which would argue for something other than "meth", eg "chain", "prop", or "field"

I still prefer "chain", as I find it intuitive, but I don't feel super strongly; I'll prob keep "chain" on my fork either way.

Cc/ @knausj85 @AndreasArvidsson @nriley any leaning here?

Reply:
No real opinion about what voice keyword to use (chain seems fine), but the list should perhaps be called common_methods, since that's what they are.

Reply:
Personally I don't separate common functions with or without `.`

Reply:
ah interesting.  so you just have them in the "funk" list with a `.` at the start?

Reply:
No. I used to have a system like this: `dot {user.code_function}: ".{code_function}"`. That way I could use all functions with or without a dot prefix. You could of course separate them into distinct lists if you want.

Reply:
We also have dot… for filename extensions in knausj. Please not "meth", thanks :-)

Reply:
Haha, fair enough re: "meth". Probably worth calling the list `code_common_method` though.

Reply:
Hmm yeh "dot" is nice. Shall we go with that?